request/api.go

```diff
@@ -0,0 +1,65 @@
+package request
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/amp-labs/cli/logger"
+	"github.com/amp-labs/cli/vars"
+)
+
+var API_VERSION = "v1" //nolint:gochecknoglobals
+
+type APIClient struct {
+	Root          string
+	ProjectId     string
+	APIKey        *string
+	RequestClient *RequestClient
+}
+
+func NewAPIClient(projectId string, key *string) *APIClient {
+	if projectId == "" {
+		// TODO: add the ability to set projectId context via a command
+		// so it doesn't need to be provided with each command.
+		logger.Fatal("Must provide a project ID in the --project flag")
+	}
+
+	return &APIClient{
+		Root:          fmt.Sprintf("%s/%s", vars.ApiURL, API_VERSION),
+		ProjectId:     projectId,
+		APIKey:        key,
+		RequestClient: NewRequestClient(),
+	}
+}
+
+type BatchUpsertIntegrationsParams struct {
+	SourceZipURL string `json:"sourceZipUrl"`
+}
+
+type Integration struct {
+	Name string `json:"name"`
+}
+
+func (c *APIClient) BatchUpsertIntegrations(
+	ctx context.Context, reqParams BatchUpsertIntegrationsParams,
+) ([]Integration, error) {
+	url := fmt.Sprintf("%s/projects/%s/integrations:batch", c.Root, c.ProjectId)
+
+	var integrations []Integration
+
+	var err error
+
+	if c.APIKey != nil && *c.APIKey != "" {
+		header := Header{Key: "X-Api-Key", Value: *c.APIKey}
+		_, err = c.RequestClient.Put(ctx, url, reqParams, &integrations, header) //nolint:bodyclose
+	} else {
+		// TODO: Default to token authentication and set Authorization header, instead of failing.
+		logger.Fatal("Must provide an API key in the --key flag")
```

Comment:
See also: [https://github.com/amp-labs/cli/blob/main/cmd/token.go](https://github.com/amp-labs/cli/blob/main/cmd/token.go)

This is how to convert a JWT login token in to a short-lived request token -- that's what you would attach in the header.

Reply:
I wish that worked 😢 It does not work for prod instance of Clerk, see https://ampersand.slab.com/posts/clerk-cli-authentication-issues-dxf397he#h9dra-issue-3-api-for-exchanging-client-token-for-session-request-token-does-not-work. That's why I implemented the API key approach for now. For MVP, we will not be using Clerk for authenticating requests from the CLI.